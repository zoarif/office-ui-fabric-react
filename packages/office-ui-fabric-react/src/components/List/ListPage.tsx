import * as React from 'react';
import {
  ExampleCard,
  IComponentDemoPageProps,
  ComponentPage,
  PageMarkdown,
  PropertiesTableSet
} from '@uifabric/example-app-base';
import { ListBasicExample } from './examples/List.Basic.Example';
import { ListGridExample } from './examples/List.Grid.Example';
import { ListScrollingExample } from './examples/List.Scrolling.Example';
import { ListGhostingExample } from './examples/List.Ghosting.Example';
import { createListItems } from '@uifabric/example-app-base';
import { ComponentStatus } from '../../demo/ComponentStatus/ComponentStatus';
import { ListStatus } from './List.checklist';

const ListBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/List/examples/List.Basic.Example.tsx') as string;
const ListGridExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/List/examples/List.Grid.Example.tsx') as string;
const ListScrollingExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/List/examples/List.Scrolling.Example.tsx') as string;
const ListGhostingExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/List/examples/List.Ghosting.Example.tsx') as string;

let _cachedItems: any;

export class ListPage extends React.Component<IComponentDemoPageProps, {}> {
  constructor(props: IComponentDemoPageProps) {
    super(props);

    _cachedItems = _cachedItems || createListItems(5000);
  }

  public render(): JSX.Element {
    return (
      <ComponentPage
        title='List'
        componentName='ListExample'
        componentUrl='https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/List'
        exampleCards={
          <div>
            <ExampleCard title='List of 5000 grid items' isOptIn={ true } code={ ListGridExampleCode }>
              <ListGridExample items={ _cachedItems } />
            </ExampleCard>
            <ExampleCard title='List of 5000 variable height items' isOptIn={ true } code={ ListBasicExampleCode }>
              <ListBasicExample items={ _cachedItems } />
            </ExampleCard>
            <ExampleCard title='Scrolling items into view' isOptIn={ true } code={ ListScrollingExampleCode }>
              <ListScrollingExample items={ _cachedItems } />
            </ExampleCard>
            <ExampleCard title='Rendering ghost items while the list is scrolling' isOptIn={ true } code={ ListGhostingExampleCode }>
              <ListGhostingExample items={ _cachedItems } />
            </ExampleCard>
          </div>
        }
        allowNativeProps={ true }
        propertiesTables={
          <PropertiesTableSet
            sources={ [
              require<string>('!raw-loader!office-ui-fabric-react/src/components/List/List.types.ts')
            ] }
          />
        }
        overview={
          <PageMarkdown>
            { require<string>('!raw-loader!office-ui-fabric-react/src/components/List/docs/ListOverview.md') }
          </PageMarkdown>
        }
        isHeaderVisible={ this.props.isHeaderVisible }
        componentStatus={
          <ComponentStatus
            { ...ListStatus }
          />
        }
      />
    );
  }
}
