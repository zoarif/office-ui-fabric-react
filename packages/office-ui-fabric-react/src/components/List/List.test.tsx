import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { List } from './List';

describe('List', () => {
  it('renders List correctly', () => {
    List.prototype.componentDidMount = jest.fn();

    const component = renderer.create(
      // tslint:disable-next-line:jsx-no-lambda
      <List items={ [] } onRenderCell={ () => null } />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});