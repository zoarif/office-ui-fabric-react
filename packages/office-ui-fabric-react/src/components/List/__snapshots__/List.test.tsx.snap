// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders List correctly 1`] = `
<div
  className="ms-List"
  role="list"
>
  <div
    className="ms-List-surface"
    role="presentation"
  />
</div>
`;
